01-node-tutorial/answers/12-http.js

```diff
@@ -0,0 +1,34 @@
+const http = require("http");
+
+const server = http.createServer((req, res) => {
+  //   if (req.url === '/') {
+  //     res.end('Welcome to our home page')
+  //   }
+  //   if (req.url === '/about') {
+  //     res.end('Here is our short history')
+  //   }
+  //   res.end(`
+  //   <h1>Oops!</h1>
+  // <p>We can't seem to find the page you are looking for</p>
+  // <a href="/">back home</a>
+  //   `)
+  // ###################################
+  // ###################################
+  //
+  //  IF YOU GET ERRORS WHILE USING ABOVE SETUP,
+  // SWITCH TO IF, ELSE IF, ELSE (BELOW)
+  // WE COVER THE CAUSE, LATER IN EXPRESS TUTORIAL
+  if (req.url === "/") {
+    res.end("Welcome to our home page");
```

Comment:
Requires change:
This file has a bug where it's crashing with an error that states that you can't "write after end". This means that once you visit `localhost:3000/` it invokes the `res.end("Welcome to our home page")`; but, instead of stopping, it's continuing down the callstack and invokes `res.end` again -- this time, it executes the "Oops, we can't seem to find the page..." code. To mitigate this error/crash, you'll want to use return statements in your conditionals. For example:
```js
    if (req.url === "/") {
        return res.end('Welcome to our home page')
    }
```

Reply:
Got it. Thanks